tests/integration/devfile/cmd_devfile_url_test.go

```diff
@@ -115,6 +115,18 @@ var _ = Describe("odo devfile url command tests", func() {
 	})
 
 	Context("Creating urls", func() {
+		It("should create a URL without port flag if only one port exposed in devfile", func() {
+			url1 := helper.RandString(5)
+			host := helper.RandString(5) + ".com"
+
+			helper.CmdShouldPass("odo", "create", "nodejs", "--project", namespace, componentName)
+
+			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), context)
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile.yaml"), filepath.Join(context, "devfile.yaml"))
+
+			helper.CmdShouldPass("odo", "url", "create", url1, "--host", host, "--secure", "--ingress")
```

Comment:
please add a check to verify that the default port was added to the devfile.

Reply:
updated